homeassistant/components/sharkiq/vacuum.py

```diff
@@ -191,6 +197,27 @@ async def async_locate(self, **kwargs: Any) -> None:
         """Cause the device to generate a loud chirp."""
         await self.sharkiq.async_find_device()
 
+    async def async_clean_room(self, rooms: list[str], **kwargs: Any) -> None:
+        """Clean specific rooms."""
+        rooms_to_clean = []
+        valid_rooms = []
+        if self.available_rooms is not None:
+            valid_rooms = self.available_rooms or []
+        for room in rooms:
+            if room in valid_rooms:
+                rooms_to_clean.append(room)
+            else:
+                raise exceptions.HomeAssistantError(
+                    "One or more of the rooms listed ("
+                    + room
+                    + ") are unavailable to your vacuum. "
+                    "Make sure all rooms match the Shark App, including capitalization."
```

Comment:
This error is meant for the user to see and should thus be translated.

Also, we don't need a stack trace for this error, so we should raise `ServiceValidationError`: https://developers.home-assistant.io/blog/2023/11/30/service-exceptions-and-translations

Here's an example of how it can be done:
https://github.com/home-assistant/core/blob/3e36556e8e4c09f71abb716f4f9f2ffc87546114/homeassistant/components/blink/camera.py#L146-L151